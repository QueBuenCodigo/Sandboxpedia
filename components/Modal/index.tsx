import React from "react";
import styled from "styled-components";

interface Props {
  visible: boolean;
  title: string;
  onCancel?: (event: React.MouseEvent<HTMLButtonElement>)=>void;
  onSubmit?: (event: React.MouseEvent<HTMLButtonElement>)=>void;
}

const Modal: React.FC<Props> = (props) => {
  const { children, visible, title, onCancel, onSubmit } = props;

  return (
    <NewModal visible={visible}>
      <ModalBody>
        <div>
          <p>{title}</p>
          <button onClick={onCancel}>X</button>
        </div>
        <div>{children}</div>
        <div>
          <button onClick={onCancel}>Cancelar</button>
          <button onClick={onSubmit}>Guardar</button>
        </div>
        </ModalBody>
    </NewModal>
  );
};

interface NewModalProps {
  visible: boolean;
}

const NewModal = styled.div<NewModalProps>`
  display: ${({ visible }) => (visible ? `block` : `none`)};
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.4);

`;

const ModalBody = styled.div`
  background-color: #fefefe;
  margin: 15% auto;
  border: 1px solid #888;
  width: 51.7%;
  display: flex;
  flex-direction: column;
  border-radius: 8px;

  div{
    

    &:nth-child(1){
      display: flex;
      flex-direction:row;
      height: 40px;
      justify-content: space-between;
      align-items: center;
      padding: 0 10px;
      border-bottom: solid 1px #f0f0f0;
      border-radius: 8px 8px 0 0;
      background-color: #3899EC;

      p{
        color: white;
        font-size: 24px
      }

      button{
        border: none;
        border-radius: 50%;
        color: white;
        background-color: #307FC3;
        outline: none;
        cursor: pointer;
      }
    }

    &:nth-child(2){
      padding: 12px;
      display:flex;
      flex-direction: column;
      justify-content: center;
    }

    &:nth-child(3){
      display: flex;
      flex-direction:row;
      height: 30px;
      justify-content: flex-end;
      padding: 0 10px;
      margin-bottom: 10px;

      button{
        border: none;
        border-radius: 20px;
        outline: none;
        color: white;
        padding: 0 20px;
        cursor: pointer;
        &:nth-child(2){
          margin-left: 5px;
          background-color: #3899EC;
          &:nth-child(2):hover{
            background-color: rgba(56,153,236,0.60);
          }
        }
        &:nth-child(1){
          margin-right: 5px;
          background-color: red;
          &:nth-child(1):hover{
            background-color: rgba(255,0,0,0.60);
          }
        }
      }
    }
  }
`;

export default Modal;
