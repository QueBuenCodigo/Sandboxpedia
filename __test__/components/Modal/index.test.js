import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Modal from '../../../components/Modal/index';

describe('<Modal/>',()=>{

    const modal=mount(<Modal/>)

    test('test del renderizado de Modal',()=>{

        expect(modal.length).toEqual(1);
    });
});



describe(' AccountListItemAccountListItem Modal',()=>{

    test('comprobar la UI del componente Modal',()=>{
        const modal= create(<Modal/>);
        expect(modal.toJSON()).toMatchSnapshot();
    });

});
 

