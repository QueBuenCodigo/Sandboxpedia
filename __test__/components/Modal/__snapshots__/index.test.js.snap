// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` AccountListItemAccountListItem Modal comprobar la UI del componente Modal 1`] = `
<div
  className="Modal__NewModal-sc-2eosl3-0 jrhpSP"
>
  <div
    className="Modal__ModalBody-sc-2eosl3-1 RWztV"
  >
    <div>
      <p />
      <button>
        X
      </button>
    </div>
    <div />
    <div>
      <button>
        Cancelar
      </button>
      <button>
        Guardar
      </button>
    </div>
  </div>
</div>
`;
